src/workerd/util/sqlite-metadata.c++

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "sqlite-metadata.h"
+
+namespace workerd {
+
+SqliteMetadata::SqliteMetadata(SqliteDatabase& db) {
+  auto q = db.run("SELECT name FROM sqlite_master WHERE type='table' AND name='_cf_METADATA'");
+  if (q.isDone()) {
+    // The _cf_METADATA table doesn't exist. Defer initialization.
+    state = Uninitialized{db};
+  } else {
+    // The metadata table was initialized in the past. We can go ahead and prepare our statements.
+    // (We don't call ensureInitialized() here because the `CREATE TABLE IF NOT EXISTS` query it
+    // executes would be redundant.)
+    state = Initialized(db);
+  }
+}
+
+kj::Maybe<kj::Date> SqliteMetadata::getAlarm() {
```

Comment:
After seeing how much `getAlarm()` is called in actor-sqlite, I'm thinking maybe it's worth caching the value in `SqliteMetadata` after all. I'd do the caching here and not in actor-sqlite as I think it'll be easier here.

Reply:
Sure, makes sense.  Yeah, caching at the SqliteMetadata layer would be a lot easier than the way I was trying to do it in ActorSqlite.

One thing I wasn't quite sure of was whether a cached value would be correct when a transaction gets rolled back.  But I suppose it would be sufficient to notify SqliteMetadata of a rollback after calling "ROLLBACK TRANSACTION" or "ROLLBACK TO..." and refresh the cached value at that point.

Reply:
Yeah, you could even just invalidate the cache so that the next time it is requested, it has to be read from sqlite again...

Reply:
Added caching/invalidation in 935e9ecd